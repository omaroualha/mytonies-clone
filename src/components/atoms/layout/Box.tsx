import { ViewProps } from "react-native";
import styled from "styled-components/native";
import {
  border,
  BorderProps,
  color,
  ColorProps,
  flexbox,
  FlexboxProps,
  layout,
  LayoutProps,
  position,
  PositionProps,
  space,
  SpaceProps,
} from "styled-system";
/**
 * Box Component
 */
export interface BoxProps
  extends ViewProps,
    SpaceProps,
    ColorProps,
    LayoutProps,
    FlexboxProps,
    PositionProps,
    BorderProps {}

export const Box = styled.View<BoxProps>`
  ${position}
  ${layout}
  ${space}
  ${flexbox}
  ${border}
  ${color}
`;
