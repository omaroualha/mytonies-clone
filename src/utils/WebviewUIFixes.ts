export const uiFixesApp = (): string => {
  const styles = `
    .main-navigation{
      display: none;
    }
    .browserupgrade .SiteTheme__Site-sc-1ycpvs-0 kKWbES .sc-ehmTmK fstLMM SiteTheme__StyledMainNavigation-sc-1ycpvs-1 cdoBQQ {
      display: none;
    }
    .b-account-page-breadcrumb-wrapper {
      display: none;
    }
    `;
  const minifiedStyles = styles.replace(/\s\s+/g, "");

  return `;(function() {
      if (document.body.dataset.injectedFromApp) return;
      document.head.insertAdjacentHTML("beforeend", "<style>${minifiedStyles}</style>");
      document.body.dataset.injectedFromApp = 'true';
    })();
    `;
};
