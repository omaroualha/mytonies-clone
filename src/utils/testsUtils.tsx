import * as React from "react";
import { BaseTheme } from "@/theme/BaseTheme";
import { NavigationContainer } from "@react-navigation/native";
import { render } from "@testing-library/react-native";
import { ThemeProvider } from "styled-components/native";

/** Render helper that renders `ui` within `NavigationContainer`. */
const renderWithNavigation = (screen: JSX.Element) => {
  return render(
    <ThemeProvider theme={BaseTheme}>
      <NavigationContainer>{screen}</NavigationContainer>
    </ThemeProvider>
  );
};

const renderWithTheme = (component: JSX.Element) => {
  return render(<ThemeProvider theme={BaseTheme}>{component}</ThemeProvider>);
};

export const testsUtils = { renderWithNavigation, renderWithTheme };
