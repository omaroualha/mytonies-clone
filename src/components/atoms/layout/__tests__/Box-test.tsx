import * as React from "react";
import { screen } from "@testing-library/react-native";
import { testsUtils } from "@/utils/testsUtils";
import { Box } from "../Box";

it(`renders Box correctly`, () => {
  testsUtils.renderWithTheme(<Box flex={1} />);
  expect(screen.toJSON()).toMatchSnapshot();
});
