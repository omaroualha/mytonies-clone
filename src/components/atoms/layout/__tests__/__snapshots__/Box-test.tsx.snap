// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Box correctly 1`] = `
<View
  flex={1}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
/>
`;
